### ISSUE TITLE
How to train the robot to avoid obstacles

### ISSUE THREAD

**Issue by Jojo11111** [intel-isl/OpenBot#119]
Hello, thanks for your great work on this project. We have problems in training the car to avoid obstacles. We would like to know that if your chairs as obstacles are always placed in the same positions  or placed randomly during each data collection process.

**Comment by thias15** [comment]
They are placed randomly during data collection. During test time they are placed in different locations. How much data did you collect? Can you share an image/video of your data collection setup?

**Comment by Jojo11111** [comment]
We collect about 6000 images with obstacles by running a python file written by ourselves. We use a game controller to control the car to avoid obstacles and record the datas just like what you did. The result is that the car runs into the obstacles without any tendency to turn around.

**Comment by thias15** [comment]
Ok, that's a very small dataset. I don't quite understand: Do you collect data with game controller or with a Python script? Can describe your data collection pipeline and maybe some sample images?

**Comment by Jojo11111** [comment]
We collect data with game controller and save the data to matched_frame_ctrl_cmd_processed.txt with a python script

Here are some frames
![2272_crop](https://user-images.githubusercontent.com/48509483/100171994-56584e80-2f02-11eb-98c4-ea4b88501e29.jpeg)

![2262_crop](https://user-images.githubusercontent.com/48509483/100171888-185b2a80-2f02-11eb-8c1a-c4fb43037884.jpeg)

And here is the matched_frame_ctrl_cmd_processed.txt
![image](https://user-images.githubusercontent.com/48509483/100172156-adf6ba00-2f02-11eb-8b8a-63478bf49bdf.png)

According to your previous reply, I think our method is correct, but the dataset is too small. We may try again later and then tell you the result.

**Comment by Jojo11111** [comment]
This time we have more images for training and testing but it still doesn't works well.
![image](https://user-images.githubusercontent.com/48509483/100303838-75c1ab00-2fd8-11eb-8051-0d3ecb0d9db0.png)

we set epoch 60 and here are the training results
![image](https://user-images.githubusercontent.com/48509483/100304062-eb2d7b80-2fd8-11eb-8b66-3e86ce10699b.png)
![image](https://user-images.githubusercontent.com/48509483/100304117-039d9600-2fd9-11eb-828d-c834d7cc0aea.png)
![image](https://user-images.githubusercontent.com/48509483/100304156-19ab5680-2fd9-11eb-8878-88e0778e582e.png)
![image](https://user-images.githubusercontent.com/48509483/100304188-262faf00-2fd9-11eb-8005-3fe92ebea2df.png)

But when the tf model was converted to tflite model, an error occurred.
![image](https://user-images.githubusercontent.com/48509483/100304524-e9b08300-2fd9-11eb-9034-86d3d92764e2.png)
We are not sure what caused this error. But when we set epoch 40, the tflite model saved successfully.
Then we used the epoch_40-model for navigation and obstacle avoidance, but the result was discouraging.
Could you give us some suggestions?

**Comment by thias15** [comment]
`[Errno 12] Cannot allocate memory` means that you do not have enough RAM memory for the conversion. I guess the memory is already at the limit during training. 

The sudden drop in performance during training is not good. Maybe the learning rate is too high. What hyperparameters do you use? (learning rate, batch size, etc.)?

**Comment by Jojo11111** [comment]
![2020-12-07 14-56-14屏幕截图](https://user-images.githubusercontent.com/48509483/101319222-932a2b00-389c-11eb-8be1-4dcdbd67aaa3.png)

**Comment by thias15** [comment]
For training the final model, it is recommended to increase the batch size. You may also try to use another model. The `model.cil_mobile` is a very compact model. You could try for example `model.pilot_net`. It has a higher capacity and may be able to learn your task more easily.

**Comment by thias15** [comment]
Closing due to inactivity. Feel free to reopen or reach out on Slack.

**Comment by Jojo11111** [comment]
Thanks you very much, we will try it later.
